ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -536,112 +175,64 @@ sub Configure_Cache {
-    return 0;
-  }
-  my @t = <CI>;
-  close(CI);
-  if ($t[0] =~ m;^([^:]+):([^:]+):(\d+)$; ) {
-    my $afsmount = $1;
-    $file_cachemount = $2;
-    $file_cache = $3;
-    if($afsmount ne "/afs") {
-      $self->error("cannot handle non-/afs mount point $afsmount in $afs_cacheinfo: ".$t[0]);
-      return 0;
-    }
+  if (open(CI, '<'.$afs_cacheinfo)) {
+      my @t = <CI>;
+      close(CI);
+      if ($t[0] =~ m;^([^:]+):([^:]+):(\d+)$; ) {
+	  $file_afsmount = $1;
+	  $file_cachemount = $2;
+	  $file_cache = $3;
+      } else {
+	  $self->error("cannot parse stored AFS cache mount or size from $afs_cacheinfo: ".$t[0]);
+	  return 0;
+      }
   } else {
-    $self->error("cannot parse stored AFS cache mount or size from $afs_cacheinfo: ".$t[0]);
+    $self->info("cannot read current AFS cachesize from $afs_cacheinfo: $! - not setting cache size");
     return 0;
   }
 
-  if(! $new_cachemount) {
-    $new_cachemount = $file_cachemount;
-  } else {
-    # all kinds of satey test for new cache mount point here
-    if (! -d $new_cachemount ) {
-      $self->warn("new AFS cache mount $new_cachemount is not a directory, ignoring");
-      $new_cachemount = $file_cachemount;
-    }
-  }
-
-  if ($new_cachemount && $file_cachemount ne $new_cachemount) {
-    $self->warn("cannot yet handle AFS cache mount move $file_cachemount -> $new_cachemount");
-  }
-
-  # Linux AFS client can autoconfigure if the AFS cache is on a
-  # separate partition (and we then won't change the other config
-  # files/run-time cache). Other machines with a separate partition
-  # will use 85% as cache size (heuristic, cache manager says 90%).
-  # Machines without a separate partition get the requested size.
-
-  # Please note that you cannot force a smaller cache if running with
-  # separate cache partition (forcing a larger one wouldn't make any sense)
-  my $dfout;
-  if (LC::Process::execute(['df', '-k', $new_cachemount],
-                           "stdout" => \$dfout,
-			   "stderr" => "stdout"
-			  ) ) {
-    my @df = split('\n', $dfout);
-
-    my @mount = grep { m{$new_cachemount} } @df;
-    if ($#mount >=0 && $mount[0] =~ m{^(.*?)\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+%)\s+(.*)}) {
-      $automatic_cache = 1;
-      my $real_cachedev = $1;
-      my $real_cachesize = $2;
-      my $real_cachemount = $6;
-      $new_cache = 0.85 * $real_cachesize;  # heuristic here
-      $self->info("found AFS cache partition on $real_cachedev, ".$real_cachesize."k");
-    } else {
-      $self->debug(2, "AFS cache $new_cachemount is not on a separate partition");
-    }
-  } else {
-    $self->warn("cannot run 'df' on the AFS cache?");
-  }
-
-  # OpenAFS-1.4/Linux takes the info from /etc/sysconfig/afs
-  if( -r $sysconfig_afs) {
-    if($automatic_cache) {
-      $new_cache = "AUTOMATIC";
+  # sanity check - don't allow cachesize bigger than 95% of a partition size
+  my @df=`df -k $file_cachemount 2>/dev/null`;
```

Comment:
@kwaegema might be nice thing to add to lib-blockdevices? (determine size from mounted FS via `df -k`)